pkg/maintenance/alertmanagerMaintenance.go

```diff
@@ -133,20 +133,21 @@ func (amm *alertManagerMaintenance) StartControlPlane(endsAt time.Time, version
 
 // Start a worker node maintenance in Alertmanager for version
 // Time is converted to UTC
-func (amm *alertManagerMaintenance) StartWorker(endsAt time.Time, version string) error {
+func (amm *alertManagerMaintenance) StartOrUpdateWorker(endsAt time.Time, version string) error {
 	comment := fmt.Sprintf("Silence for OSD worker node upgrade to version %s", version)
 	mList, err := amm.client.List([]string{})
 	if err != nil {
 		return err
 	}
-	exists := silenceExists(mList, comment)
-	if exists {
-		return nil
-	}
+	exists, silence := silenceExists(mList, comment)
```

Comment:
Just a suggestion to add a `Filter` func to `alertManagerSilenceClient`
```.go
type silenceFilterPredicate func(*amv2Models.GettableSilence) bool

// list silences in Alertmanager instance defined in transport
func (ams *alertManagerSilenceClient) Filter(predicate silenceFilterPredicate) (*[]amv2Models.GettableSilence, error) {
	silences, err := ams.List([]string{})
	if err != nil {
		return nil, err
	}

	filteredSilences := []amv2Models.GettableSilence{}
	for _, s := range silences.Payload {
		if predicate(s) {
			filteredSilences = append(filteredSilences, *s)
		}
	}

	return &filteredSilences, nil
}
```
Up to you if you think its worth it